doc/widgets/singlemode.md

```diff
@@ -0,0 +1,50 @@
+Single Mode
+===========
+
+Convert multimodal graphs to single modal.
+
+**Inputs**
+
+- Network: An instance of a bipartite network graph.
+
+**Outputs**
+
+- Network: An instance of single network graph.
+
+**Single Mode** works with bipartite networks, where subsets are marked with a binary discrete attribute. The widget outputs a network where nodes represent the selected value (e.g. persons) and edges the remaining value (e.g. event). The widget also works with non-binary discrete attributes, but will treat them as binary (one-versus-all).
+
+![](images/Single-Mode-stamped.png)
+
+1. Mode indicator:
+   - *Feature*: discrete feature labeling network subsets.
+   - *Connect*: value used as nodes.
+   - *by*: value used as edges.
+2. Edge weights: compute weights for the output network.
+3. Information on the output network.
+
+####Edge weights
+
+![](images/Single-Mode-weights.png)
+
+- *No weights*: all weights are set to 1.
+- *Number of connections*: weights correspond to the number of common connections (e.g. events the two people participated in).
+- *Weighted number of connections*
+- *Geometric normalization*
+- *Geometric normalization by degrees*
+- *Normalization by sum of input weights*
+- *Normalization by sum of output weights*
+- *Normalization by minimal sum of weights*
+- *Normalization by maximal sum of weights*
```

Comment:
@janezd Most of the above is missing, because I wasn't able to understand what these functions do. Assistance appreciated.

Reply:
I started writing this:
"""
- *Weighted number of connections*: for every common event, the widget computes the product of weights that connect each person with the event; the weights of the edge between the two persons equals the sum of these products.

- *Geometric normalization*: like above, except that each product is normalized by dividing it with the geometric mean of the sum of weights of edges of the two nodes.

- *Geometric normalization by degrees*: like above, except that normalization is done by node degrees, not sums of weights.
"""

... but then it gets worse and worse. One should use formulas for this. I think we should refer the user to literature. We can describe the first few (perhaps what I've already written above) and then say "For details and for description of other options, see Vlado Batagelj's [Introduction to Network Science using Pajek - section 7](http://vladowiki.fmf.uni-lj.si/lib/exe/fetch.php?media=pajek:ev:pdf:na07.pdf), page 38 (Normalizations)